### PR TITLE
Improve the ./easy-setup.sh script

### PR DESCRIPTION
**Pull request by yodapotatofly**
- **not use docker group anymore (bad practice) -> run script as root instead**

The use of "docker" group is considered bad practice because, as docker engine runs as root, it gives root privileges to any user in the group. Unlike sudo, there is no password check on the docker command so this could lead to security issues.

The use of the docker group also made the script less conveniant since the user had to log off and in again for the new group to be taken into account.

- **Command line options for installing docker, compose and portainer in non interactive mode**

Following PR #343 , it can be usefull to be able to install components (docker, compose or portainer) during a non interactive run of the script. Therefore, I'm introducing `--install-docker`,  `--install-docker-compose`,  `--install-portainer` and `--install-all`
 options.
 When used in interactive mode, the user will not be prompted for those either.
 
 
- **update disclaimer with tested linux**

Updating the initial disclaimer with the distros and versions the script has been tested on

### CODE REVIEW

**Comment by pevma** [comment]
Can you please adjust the commit msg: `short: longer`.
Do we need doc update with those?

**Comment by pevma** [comment]
When using a command like that:
```
./easy-setup.sh  --non-interactive -i tppdummy0 --iA  --es-memory 4G  && docker-compose up -d
```
the script gets into infinite loop when it comes to the interface - if that interface is not up or available/present. Might worth to put in a sanity check to be on the safe side.

**Comment by pevma** [comment]
Also when spun up , If you open Scirius Hunt, after you click on a couple of filter sets it reverts back to the login page.

File: docker/easy-setup.sh [hunk 1]
```diff
@@ -41,15 +41,25 @@ function Help(){
...
+    echo -e " --iP,--install-portainer"
+    echo -e "       Install portainer\n"
+    echo -e " --iA,--install-all"
+    echo -e "       equivalent to \"-iD -iC -iP\", install docker, docker-compose and portainer\n"
+    echo -e " -s,--skip-checks"
+    echo -e "       Run the scirpt without checking if docker and docker-compose are installed. Use this only if you know that both docker and docker-compose are already installed with proper versions. Otherwise, the script will probably fail\n"
     echo -e " --scirius-version <version>"
     echo -e "       Defines the version of scirius to use. The version can be a branch name, a github tag or a commit hash. Default is 'master'\n"
     echo -e " --elk-version <version>"
     echo -e "       Defines the version of the ELK stack to use. Default is '7.12.0'. The version should match a tag of Elasticsearch, Kibana and Logstash images on the dockerhub\n"
```
Comment (pevma) [inline docker/easy-setup.sh:57]: Default version currently is 7.15.1

**Review by pevma — COMMENTED** [review]
Actually disregard the previous comment (Scirius above) - I had an existing localhost proxy interfering, sorry about the noise.

**Comment by yodapotatofly** [comment]
> When using a command like that:
> ```
> ./easy-setup.sh  --non-interactive -i tppdummy0 --iA  --es-memory 4G  && docker-compose up -d
> ```
> the script gets into infinite loop when it comes to the interface - if that interface is not up or available/present. Might worth to put in a sanity check to be on the safe side.

fixed

> Default version currently is 7.15.1

fixed

**Comment by pevma** [comment]
One thing to consider is that ES memory can only be done during the first time setup. If you do composer down then up again, the installation will default to 2gb. Is my understanding correct ?

**Comment by pevma** [comment]
My bad on my previous comment - it will get written into `.env`.

**Pull request merged**